### ISSUE TITLE
waf dist fails with broken symlinks

### ISSUE THREAD

**Issue by rasky** [waf-project/waf#1719]
It looks like `waf dist` fails with an exception if the source tree contains a broken symlink. I would expect it to just pack the symlink the way it is (like git does), without trying to resolve it.

```
Traceback (most recent call last):
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 114, in waf_entry_point
    run_commands()
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 175, in run_commands
    ctx=run_command(cmd_name)
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 166, in run_command
    ctx.execute()
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 240, in execute
    self.archive()
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 257, in archive
    self.add_tar_file(x,tar)
  File "<myproject>/.waf-1.8.20-c859ca7dc3693011756f4edf45c36626/waflib/Scripting.py", line 288, in add_tar_file
    fu=open(p,'rb')
IOError: [Errno 2] No such file or directory: '<myproject>/path/broken-symlink.so'
```

**Comment by ita1024** [comment]
This is not supported. Create [a command that overrides](https://github.com/waf-project/waf/blob/master/wscript#L411) "dist" and [reimplement the method `add_tar_file`](https://github.com/waf-project/waf/blob/master/waflib/Scripting.py#L415).

**Comment by ita1024** [comment]
Or use 41dcf329b06d503c68dd7b48f38e4cff009d3fcf in the branch waf-1.9

**Comment by ita1024** [comment]
The fix is in waf 1.9.0
